sdk/main/include/AccountAllowanceDeleteTransaction.h

```diff
@@ -17,136 +17,107 @@
-   * Validate the checksums.
+   * Construct from a TransactionBody protobuf object.
    *
-   * @param client The client with which to validate the checksums
+   * @param transactionBody The TransactionBody protobuf object from which to construct.
+   * @throws std::invalid_argument If the input TransactionBody does not represent a CryptoDeleteAllowance transaction.
    */
-  virtual void validateChecksums(const Client& client) const override;
+  explicit AccountAllowanceDeleteTransaction(const proto::TransactionBody& transactionBody);
 
   /**
-   * Derived from Transaction. Called in freezeWith(Client) just before the
-   * transaction body is built. The intent is for the derived class to assign
-   * their data variant to the transaction body.
+   * Remove all NFT allowances from an account.
    *
-   * @param body The transaction's body part. (XXX)
+   * @param nftId The ID of the NFT to remove as an allowance.
+   * @param owner The ID of the account from which to remove the allowances.
+   * @return A reference to this AccountAllowanceDeleteTransaction object with the added allowance removal.
+   * @throws IllegalStateException If this AccountAllowanceDeleteTransaction is frozen.
    */
-  virtual void onFreeze(proto::TransactionBody* body) const override;
+  AccountAllowanceDeleteTransaction& deleteAllTokenNftAllowances(const NftId& nftId, const AccountId& owner);
 
   /**
-   * Called in schedule() when converting transaction into a scheduled version.
+   * Get the list of a NFT allowance removals added to this AccountAllowanceDeleteTransaction.
    *
-   * @param body The schedulable transaction's body part. (XXX)
+   * @return The list of a NFT allowance removals added to this AccountAllowanceDeleteTransaction.
    */
-  virtual void onScheduled(proto::SchedulableTransactionBody* body) const override;
-
-  /**
-   * Build an account allowance delete protobuf message based on the data in
-   * this class.
-   *
-   * @return An account allowance delete protobuf message.
-   */
-  proto::CryptoDeleteAllowanceTransactionBody* build() const;
+  [[nodiscard]] inline std::vector<TokenNftRemoveAllowance> getTokenNftAllowanceDeletions() const
+  {
+    return mNftAllowanceDeletions;
+  }
 
+private:
   /**
-   * Delete all NFT allowances.
+   * Derived from Executable. Construct a Transaction protobuf object from this AccountAllowanceDeleteTransaction
+   * object.
    *
-   * @param nftId            The token's ID.
-   * @param ownerAccountId   The owner's account ID.
-   * @return Reference to this AccountAllowanceDeleteTransaction object
+   * @param client The Client trying to construct this AccountAllowanceDeleteTransaction.
+   * @param node   The Node to which this AccountAllowanceDeleteTransaction will be sent. This is unused.
+   * @return A Transaction protobuf object filled with this AccountAllowanceDeleteTransaction object's data.
+   * @throws UninitializedException If the input client has no operator with which to sign this
```

Comment:
Yeah, the request is made into a `proto::Transaction`, which contains bytes that are signed by the client's operator. If the client has no operator, it can't be signed, and thus will throw the exception.